app/views/contexts/summary_context.py

```diff
@@ -1,316 +1,146 @@
-    section_id = schema.get_section_id_for_block_id(current_location.block_id)
-    group = schema.get_group_for_block_id(current_location.block_id)
-    blocks = []
-    answers_to_calculate = rendered_block['calculation']['answers_to_calculate']
-    blocks_to_calculate = [
-        schema.get_block_for_answer_id(answer_id) for answer_id in answers_to_calculate
-    ]
-    unique_blocks = list({block['id']: block for block in blocks_to_calculate}.values())
-
-    for block in unique_blocks:
-        if block['type'] == 'Question':
-            transformed_block = _remove_unwanted_questions_answers(
-                block,
-                answers_to_calculate,
-                answer_store,
-                list_store,
-                metadata,
-                schema,
-                current_location=current_location,
-            )
-            if set(get_answer_ids_in_block(transformed_block)) & set(
-                answers_to_calculate
-            ):
-                blocks.append(transformed_block)
+        return context
 
-    return [{'id': section_id, 'groups': [{'id': group['id'], 'blocks': blocks}]}]
+    def build_context_for_section(self, section_id, list_name=None, list_item_id=None):
+        """
+        Builds the context for a single section
+        Args:
+            section_id: The id of the target section
+            list_item_id: If this is a repeating section, the list_item_id
+        """
 
+        groups = self.build_groups_for_section(section_id, list_name, list_item_id)
 
-def _remove_unwanted_questions_answers(
-    block,
-    answer_ids_to_keep,
-    answer_store,
-    list_store,
-    metadata,
-    schema,
-    current_location,
-):
-    """
-    Evaluates questions in a block and removes any questions not containing a relevant answer
-    """
-    block_question = choose_question_to_display(
-        block,
-        schema,
-        answer_store,
-        list_store,
-        metadata,
-        current_location=current_location,
-    )
+        context = {'summary': {'groups': groups, 'answers_are_editable': True}}
 
-    reduced_block = block.copy()
+        return context
 
-    matching_answers = []
-    for answer_id in answer_ids_to_keep:
-        matching_answers.extend(schema.get_answers_by_answer_id(answer_id))
+    def section_summary(self, current_location):
+        section_id = current_location.section_id
+        section = self._schema.get_section(current_location.section_id)
+        list_item_id = current_location.list_item_id
+        list_name = current_location.list_name
+        block = self._schema.get_block(current_location.block_id)
 
-    questions_to_keep = [answer['parent_id'] for answer in matching_answers]
+        context = self.build_context_for_section(section_id, list_name, list_item_id)
+        context['summary']['collapsible'] = block.get('collapsible', False)
+        context['summary']['summary_type'] = 'SectionSummary'
```

Comment:
`build_context_for_section` is only called from here. It would be clearer to just inline that method here:
```python
groups = self.build_groups_for_section(section_id, list_name, list_item_id)
context = {
    'summary': {
        'groups': groups, 
        'answers_are_editable': True,
        'collapsible': block.get('collapsible', False),
        'summary_type': 'SectionSummary',
    }
}
```

Reply:
Done